model/task/task.go

```diff
@@ -4180,3 +4221,114 @@ func (t *Task) SetSortingValueBreakdownAttributes(ctx context.Context, breakdown
 	)
 	t.SortingValueBreakdown = breakdown
 }
+
+// CalculateOnDemandCost calculates the on-demand cost of running a task.
+func CalculateOnDemandCost(runtimeSeconds float64, distroCostData distro.CostData, financeConfig evergreen.CostConfig) float64 {
+	if runtimeSeconds <= 0 {
+		return 0
+	}
+
+	return runtimeSeconds * (distroCostData.OnDemandRate / 3600.0) * (1 - financeConfig.OnDemandDiscount)
+}
+
+// CalculateAdjustedTaskCost calculates the adjusted cost of running a task based on the provided runtime,
+// distro cost data, and admin finance settings using the finance formula.
+func CalculateAdjustedTaskCost(runtimeSeconds float64, distroCostData distro.CostData, financeConfig evergreen.CostConfig) float64 {
+	if runtimeSeconds <= 0 {
+		return 0
+	}
+
+	savingsPlanComponent := financeConfig.FinanceFormula * (distroCostData.SavingsPlanRate / 3600.0) * (1 - financeConfig.SavingsPlanDiscount)
+	onDemandComponent := (1 - financeConfig.FinanceFormula) * (distroCostData.OnDemandRate / 3600.0) * (1 - financeConfig.OnDemandDiscount)
+	costPerSecond := savingsPlanComponent + onDemandComponent
+
+	return runtimeSeconds * costPerSecond
+}
+
+// CalculateTaskCost calculates both on-demand and adjusted costs for a task.
+func CalculateTaskCost(runtimeSeconds float64, distroCostData distro.CostData, financeConfig evergreen.CostConfig) TaskCost {
+	return TaskCost{
+		OnDemandCost: CalculateOnDemandCost(runtimeSeconds, distroCostData, financeConfig),
+		AdjustedCost: CalculateAdjustedTaskCost(runtimeSeconds, distroCostData, financeConfig),
+	}
+}
+
+func isFinanceConfigured(financeConfig evergreen.CostConfig) bool {
```

Comment:
Would it be cleaner if we use struct receivers for this function and the two below it?